### DISCUSSION TITLE
`runX` and `handleX` in effect modules

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8694] [Ideas]
currently we have `runX: Unit -> t \ ef -> t \ ef - X`. This is nice for the simple case, but when you have multiple effects, they are hard to compose `runA(() -> runB(() -> body))`. so I propose that we have the base handler function `handleX: a -> b \ ef -> a -> b \ ef - X` where `runX` can then be a shorthand convenience method

**Comment by magnus-madsen** [comment]
I agree.

**Reply by JonathanStarup** [reply]
#8702

Do we close this discussion now?

**Reply by magnus-madsen** [reply]
I suppose so.
